src/lib/chawathe98_matcher.rs

```diff
@@ -681,9 +681,9 @@ pub fn edgecover_solver(
     // We would check through the edge keep and only retain edges in edge keep;
     let mut edge_filtered: Vec<Edge> = Vec::new();
     for edge in new_matcher_pruning.list_edges.borrow().iter() {
-        if edge_keep.contains(&(edge.from_node, edge.to_node, edge.edge_type, edge.value)) {
-            edge_filtered.push(Edge::new(edge.from_node, edge.to_node, edge.value, edge.edge_type));
-        } else if edge.edge_type == EdgeType::DELETE {
+        if edge_keep.contains(&(edge.from_node, edge.to_node, edge.edge_type, edge.value))
+           || edge.edge_type == EdgeType::DELETE
```

Comment:
Wouldn't it be worth having the simple case first? I don't know how LLVM chooses to order these, but it might choose to keep the order unchanges -- and the boolean check is bound to be faster.